### PR TITLE
✨  use knex-migrator

### PR DESCRIPTION
**Pull request by kirrg001**
### TODO
- [x] test this code
- [x] knex-migrator package.json
- [x] forward path to knex migrator where to find the .knex-migrator file

Check database health before starting Ghost and run `knex-migrator` init if database was never initialised.

### CODE REVIEW

File: lib/commands/start.js [hunk 1]
```diff
@@ -45,6 +46,13 @@ module.exports = BaseCommand.extend({
         // TODO: rethink this
         return this.runCommand('config', 'paths.contentPath', path.join(process.cwd(), 'content'), {environment: environment})
             .then(function () {
+                knexMigrator = new KnexMigrator();
```
Comment (acburdine) [inline lib/commands/start.js]: I know this is still a WIP - but in order to improve load times of the CLI, there's a better way to have the requires work:

```js
var KnexMigrator = require('knex-migrator'),
    knexMigrator = new KnexMigrator();

// ... do other things
```

If you do this and remove the two changes above this then the code changes look good to me - will test it locally once it's out of WIP status :smile:

**Comment by kirrg001** [comment]
@acburdine the new Ghost-CLI version won't run with an older Ghost alpha version when merging this PR

**Comment by acburdine** [comment]
@kirrg001 I went ahead and changed the code to what I was asking (because needing to get this merged) - but here's why:

The cli has to require each command file (so it can get all of the necessary variables like command name, arguments, options, etc). Because of this, adding a bunch of require statements to the top of the file increases load times, especially when the command is not even being run (for example, if you run `ghost config` - you don't even need `ghost start` or its dependency on "knex-migrator", yet with your code knex-migrator would still be loaded).

Basically, this is probably over-ambitious optimization, but as it was a quick change and the same practice is followed everywhere else in the code, I think it's necessary.

I'm willing to revert this change in the future if there's good reason to, though 😄

File: lib/commands/start.js (hunk 1 above)
Comment (acburdine) [inline lib/commands/start.js]: see below comment - I see why you didn't instantiate knex-migrator until later on in the code, but at the same time moving the variables down makes more perf sense imo

**Comment by acburdine** [comment]
So I pulled this locally to test - and it's not working - neither sqlite nor mysql dependencies are installed, so ghost start doesn't work. And I'm not sure that having ghost-cli depend on both sqlite and mysql is a good idea... :confused:

**Comment by kirrg001** [comment]
> So I pulled this locally to test - and it's not working - neither sqlite nor mysql dependencies are installed, so ghost start doesn't work. And I'm not sure that having ghost-cli depend on both sqlite and mysql is a good idea... 😕

`ghost start` might not work because of a missing PR in Ghost?
https://github.com/TryGhost/Ghost/pull/7585/files
